opendatahub-io/odh-dashboard
# Use Dashboard SA Token to access prometheus GPU data.

Resolves: #274

Testing instructions(WIP):

1. Add following label to the opendatahub namespace: `openshift.io/cluster-monitoring: 'true'`
2. Deploy mock-server.yaml and service-monitor.yaml from the dev overlay to create a mock metric exporter. (It is also required to include this role binding: 
```
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: odh-dashboard-monitoring
subjects:
  - kind: ServiceAccount
    name: odh-dashboard
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-monitoring-view 
```
)

4. Deploy a dashboard image with changes from this PR (`quay.io/mroman_redhat/odh-dashboard:dev-10` contains the latest changes)
5. Access the /api/gpu endpoint
6. The endpoint should respond with 2 available GPUs
7. Alternatively, the spawner UI should also show the correct number of GPUs in the GPU dropdown.
